bin/oneoff/hoai_2025/README.md

```diff
@@ -0,0 +1,99 @@
+# Hour of AI 2025 Generation Scripts
+
+## Music
+
+The `generate_music` script will contact AI (gemini) to produce musiclab
+psuedo-code for the various possible packs and choice combinations a student
+might make during the activity.
+
+There are two types of 'adlib' prompts:
+
+```
+Code a music mix that layers {layer1} and {layer2} together.
+```
+
+Where `layer\d` are one of the following (including duplicates): beats,
+vocals, leads, bass. These correspond to the sound types that exist within
+music lab packs. The scripts also can handle a third layer and a length
+which was an early version of this prompt (see `:use_full_layers` option
+hard-coded in the script)
+
+```
+Code a {mood} music mix with a {length} length, using {drums} drums.
+```
+
+Where the 'mood' is either simple, creative, or wild and the drums correspond
+to the various provided packs and a 'default' option which uses the drums of
+the current pack (song) which a student selects when first viewing the level.
+
+### Usage
+
+To generate ALL prompts (yikes... it's like 20k AI responses and that will
+cost a relative ton), use this command:
+
+```
+./bin/oneoff/hoai_2025/generate_music --produce-layers --produce-moods
+```
+
+Likely you'll want to generate just the prompts and save them where the
+final items would ultimately go: (This is a great place to start because
+it generates _something_ which prevents any _real_ runs from sending any
+AI prompts)
+
+Again, this does **not** send any prompts to actual AI agents.
+
+```
+./bin/oneoff/hoai_2025/generate_music --produce-layers --produce-moods --just-save-prompts
+```
```

Comment:
Slightly confused: does it generate prompts and save them as files because they are something interesting to examine?